delta/src/main/scala/io/qbeast/spark/delta/DeltaQbeastSnapshot.scala

```diff
@@ -46,8 +49,9 @@ case class DeltaQbeastSnapshot(tableID: QTableID) extends QbeastSnapshot with De
 
   override val basePath: Path = new Path(tableID.id)
 
-  override val snapshot: Snapshot =
-    DeltaLog.forTable(SparkSession.active, tableID.id).update()
+  private val deltaLog: DeltaLog = DeltaLog.forTable(SparkSession.active, tableID.id)
```

Comment:
To be honest, I'm not super happy with the Snapshot management in `QbeastSnapshot`... We are always loading the newest version, while the `QbeastSnapshot` should have a 1:1 relation with an underlying Snapshot of the Table.

QbeastTable can have access to different QbeastSnapshots. But a `QbeastSnapshot` must be isolated. Always access to the same set of files and metadata. 

I know we changed that during the Refactor because we were anyways initializing the object with the most updated one, but I think we should look at it carefully and fix it before adding more changes to it...

what do you think? feels relevant? or would you like to fix it after?

Reply:
From my point of view, currently I'm only adding things that I need for hudi. 
I think we should create a separate issue for this and address it in a different PR.